src/test/CdpRegistry.t.sol

```diff
@@ -0,0 +1,105 @@
+
+pragma solidity 0.6.12;
+
+import "../CdpRegistry.sol";
+import "./TestBase.sol";
+
+contract Usr {
+    CdpRegistry cdpRegistry;
+
+    constructor(CdpRegistry cdpRegistry_) public {
+        cdpRegistry = cdpRegistry_;
+    }
+
+    function open(bytes32 ilk, address usr) public returns (uint256){
+        return cdpRegistry.open(ilk, usr);
+    }
+}
+
+contract MockCdpManager {
+    uint256 public cdpi;
+    mapping (uint256 => address) public owns; // CDPId => Owner
+    mapping (uint256 => bytes32) public ilks; // CDPId => Ilk
+
+    function open(bytes32 ilk, address usr) public returns (uint256) {
+        cdpi = cdpi + 1;
+        owns[cdpi] = usr;
+        ilks[cdpi] = ilk;
+        return cdpi;
+    }
+}
+
+contract CdpRegistryTest is TestBase {
+
+    MockCdpManager cdpManager;
+    CdpRegistry    cdpRegistry;
+    Usr            usr1;
+    Usr            usr2;
+
+    function setUp() public virtual {
+        cdpManager  = new MockCdpManager();
+        cdpRegistry = new CdpRegistry(address(cdpManager));
+        usr1        = new Usr(cdpRegistry);
+        usr2        = new Usr(cdpRegistry);
+    }
+
+    function test_open_different_ilk_same_usr() public {
+        uint256 cdp = usr1.open("ILK1", address(usr1));
+        assertEq(cdp, 1);
+        assertEq(cdpRegistry.ilks(1), "ILK1");
+        assertEq(cdpRegistry.owns(1), address(usr1));
+        assertEq(cdpRegistry.cdps("ILK1", address(usr1)), 1);
+        assertEq(cdpManager.ilks(1), "ILK1");
+        assertEq(cdpManager.owns(1), address(cdpRegistry));
+
+        cdp = usr1.open("ILK2", address(usr1));
+        assertEq(cdp, 2);
+        assertEq(cdpRegistry.ilks(2), "ILK2");
+        assertEq(cdpRegistry.owns(2), address(usr1));
+        assertEq(cdpRegistry.cdps("ILK2", address(usr1)), 2);
+        assertEq(cdpManager.ilks(2), "ILK2");
+        assertEq(cdpManager.owns(2), address(cdpRegistry));
+    }
+
+    function test_open_same_ilk_different_usr() public {
+        uint256 cdp = usr1.open("ILK1", address(usr1));
+        assertEq(cdp, 1);
+        assertEq(cdpRegistry.ilks(1), "ILK1");
+        assertEq(cdpRegistry.owns(1), address(usr1));
+        assertEq(cdpRegistry.cdps("ILK1", address(usr1)), 1);
+        assertEq(cdpManager.ilks(1), "ILK1");
+        assertEq(cdpManager.owns(1), address(cdpRegistry));
+
+        cdp = usr2.open("ILK1", address(usr2));
+        assertEq(cdp, 2);
+        assertEq(cdpRegistry.ilks(2), "ILK1");
+        assertEq(cdpRegistry.owns(2), address(usr2));
+        assertEq(cdpRegistry.cdps("ILK1", address(usr2)), 2);
+        assertEq(cdpManager.ilks(2), "ILK1");
+        assertEq(cdpManager.owns(2), address(cdpRegistry));
+    }
+
+    function testFail_open_same_ilk_same_usr() public {
+        uint256 cdp = usr1.open("ILK1", address(usr1));
+        cdp = usr1.open("ILK1", address(usr1));
+    }
+
+    function testFail_open_for_other_usr() public {
+        uint256 cdp = usr1.open("ILK1", address(usr2));
```

Comment:
```suggestion
        usr1.open("ILK1", address(usr2));
```
Unused variable

Reply:
There is also a warning on the charter function of `flee`. Not related to this PR but if can be fixed here probably good option.

Reply:
Fixed those, didn't notice you suggested the change here.

Reply:
Hmm I still see this with the issue.

Reply:
Sorry added only one of the files, pls check now.

Reply:
Looks good